templates/default/net/netmap.html

```diff
@@ -0,0 +1,239 @@
+        </td>
+    </tr>
+    </tbody>
+</table>
+
+<style>
+    .lmsbox_mark_odd_rows tbody tr:nth-child(odd){
+        background-color: #EBE4D6;
+    }
+    .lmsbox_mark_odd_rows tbody td {
+        padding-left: 10%;
+    }
+    .lmsbox_mark_odd_rows tbody td:hover {
+        background-color: #CCFFCC;
+        cursor: pointer;
+    }
+    .node:before   { content:url('img/node.gif');   }
+    .netdev:before { content:url('img/netdev.gif'); }
+</style>
+<table class="lmsbox lmsbox_mark_odd_rows" id="ip_addr_container" style="display: none;">
+    <thead>
+        <tr>
+            <td colspan="4" class="bold">{trans('Result')}</td>
+        </tr>
+    </thead>
+    <tbody></tbody>
+</table>
+
+<script>
+    var ip_end     = 0;
+    var ip_current = 0;
+    var data;
+
+    $( function() {
+        /*!
+         * \brief Send request after press ENTER on IP address input.
+         */
+        $('#ipaddr').keydown( function(key) {
+            if (key.which === 13) {
+                $('#do_search').trigger('click');
+            }
+        });
+
+        /*!
+         * \brief Get used IP addresses.
+         */
+        $('#do_search').click( function() {
+            if ( $('#ipaddr').val() == '' ) {
+                return;
+            }
+
+            $('#ip_addr_container').fadeOut(100, function(){
+                // clear current table rows
+                $('#ip_addr_container tbody tr').remove();
+
+                // load data
+                getAddressList($('#ipaddr').val(), $('#mask_select').val());
+
+                // show table
+                $('#ip_addr_container').fadeIn(300);
+            });
+        });
+
+        /*!
+         * \brief Append IP addresses while scroll down.
+         */
+        $(window).scroll(function() {
+            if ( $(this).scrollTop() + $(this).height() > $(document).height() - 500 ) {
+                tabFill();
+            }
+        });
+
+        /*!
+         * \brief On node table cell click.
+         */
+        $(document).on('click', '.node', function() {
+            window.location.href = "?m=nodeinfo&id=" + $(this).attr('data-id');
+        });
+
+        /*!
+         * \brief On net device table cell click.
+         */
+        $(document).on('click', '.netdev', function() {
+            window.location.href = "?m=netdevinfo&id=" + $(this).attr('data-id');
+        });
+    });
+
+    /*!
+     * \brief Returns ip address usage list.
+     *
+     * \param string ip   Ip address
+     * \param string mask Network mask
+     */
+    function getAddressList(ip, mask) {
+        $('#ip_addr_container').css('display', 'none');
+
+        $.ajax({
+            url: "?&m=netmap&ip=" + ip + "&mask=" + mask
+        }).done( function( response ) {
+            data = JSON.parse(response);
+
+            ip_current = data['ip_start'];
+            ip_end     = data['ip_end'];
+
+            tabFill();
+        });
+    }
```

Comment:
Czy to zapytanie ajax nie pójdzie czasem synchronicznie generując ostrzeżenia w przeglądarkach?